server.go

```diff
@@ -102,6 +103,24 @@ type Response struct {
 	ErrType responseErr
 }
 
+// Permissions maps "service.method" of a peer to boolean permission
+type Permissions map[peer.ID]map[string]bool
+
+// AuthorizeWithMap returns an authrorization function that follows the authorization strategy as described in the
+// given map(maps "service.method" of a peer to boolean permission)
+func AuthorizeWithMap(p Permissions) func(peer.ID, string, string) bool {
+	if p != nil {
+		return func(pid peer.ID, svc string, method string) bool {
+			return p[pid][svc+"."+method]
+		}
+	}
+
+	// If permission map is nil, no method would be allowed
+	return func(pid peer.ID, svc string, method string) bool {
```

Comment:
the construct here is weird. Better to:

```go
		return func(pid peer.ID, svc string, method string) bool {
                        if p == nil { return false }
			return p[pid][svc+"."+method]
		}
```

Reply:
That said you also need to check that `p[pid]` is not nil

Reply:
I am not sure we need to alias the map as separate `Permissions` type. Opinion @lanzafame ?

Reply:
I agree that we don't need the alias. 

> That said you also need to check that p[pid] is not nil

Accessing `p[pid][svc+"."+method]` when `p[pid]` is nil will return false, which is the appropriate result anyway.

Reply:
doesn't it panic?

Reply:
Nope

Reply:
@hsanjuan https://play.golang.org/p/J3j4PLevxv6

Reply:
And https://play.golang.org/p/FmZ28_uxByF

Reply:
https://play.golang.org/p/DfhdeyZnG9Z you're right. How many lines of code have I wasted checking if maps are nil ?

Actually, since assignment to nil maps does panic I always thought retrieval would do too :/

Reply:
> A nil map behaves like an empty map when reading, but attempts to write to a nil map will cause a runtime panic; don't do that. 
Source: https://blog.golang.org/go-maps-in-action